packages/tools/devtools/devtools-test-app/src/FluidObject.ts

```diff
@@ -3,14 +3,102 @@
  * Licensed under the MIT License.
  */
 
-import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct/internal";
+import {
+	DataObject,
+	DataObjectFactory,
+	PureDataObjectFactory,
+	TreeDataObject,
+} from "@fluidframework/aqueduct/internal";
 import { SharedCell } from "@fluidframework/cell/internal";
 import type { IFluidHandle, IFluidLoadable } from "@fluidframework/core-interfaces";
 import { SharedCounter } from "@fluidframework/counter/internal";
 import { SharedMatrix } from "@fluidframework/matrix/internal";
 import { SharedString } from "@fluidframework/sequence/internal";
 import { type ITree, SchemaFactory, TreeViewConfiguration } from "@fluidframework/tree";
-import { SharedTree } from "@fluidframework/tree/internal";
+import { SharedTree, type TreeView } from "@fluidframework/tree/internal";
+import { v4 as uuid } from "uuid";
+
+import { TodoList, TodoItem } from "./Schema.js";
+
+/**
+ * Props used when creating a new todo item.
+ */
+interface TodoItemProps {
+	/**
+	 * The initial text to populate the todo item's title with.
+	 * This value will be inserted into the shared string at index 0.
+	 */
+	readonly startingText: string;
+}
+
+/**
+ * Object so we can test {@link TreeDataObject} inside {@link AppData}.
+ */
+export class AppDataTree extends TreeDataObject<TreeView<typeof TodoList>> {
+	public static readonly Name = "@devtools-example/test-app-tree-data-object";
+
+	private readonly config = new TreeViewConfiguration({ schema: TodoList });
+	private static readonly factory = new PureDataObjectFactory<
+		TreeDataObject<TreeView<typeof TodoList>>
+	>(AppDataTree.Name, AppDataTree, [SharedString.getFactory(), SharedTree.getFactory()], {});
+
+	public static getFactory(): PureDataObjectFactory<
+		TreeDataObject<TreeView<typeof TodoList>>
+	> {
+		return AppDataTree.factory;
+	}
+
+	/**
+	 * Converts the underlying ITree into a typed TreeView using the provided schema configuration.
+	 *
+	 * @param tree - The ITree instance to view.
+	 * @returns A typed TreeView using the TodoList schema.
+	 */
+	public override generateView(tree: ITree): TreeView<typeof TodoList> {
+		return tree.viewWith(this.config) as unknown as TreeView<typeof TodoList>;
+	}
+
+	/**
+	 * Initializes the tree with a default title and empty todo item list.
+	 * @remarks Called during the initial creation of the data object.
+	 */
+	public override async initializingFirstTime(): Promise<void> {
+		const title = SharedString.create(this.runtime);
+		title.insertText(0, "Title");
+
+		this.treeView.initialize(new TodoList({ title: title.handle, items: [] }));
+	}
+
+	/**
+	 * Adds a new todo item to the list.
+	 *
+	 * @param props
+	 * -`startingText`: The text to prefill into the item's title.
```

Comment:
Nit: the `startingText` property is already documented in the `TodoItemProps` interface. I think you can remove this block.
```suggestion
```